Documentation/ApiOverview/TSFE/Index.rst

```diff
@@ -0,0 +1,171 @@
+.. include:: /Includes.rst.txt
+.. index:: TSFE; TypoScriptFrontendController
+.. _tsfe:
+
+====
+TSFE
+====
+
+..  contents::
+    :local:
+
+What is TSFE?
+=============
+
+TSFE is short for :php:`\TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController`,
+a class which exists in the system extension EXT:frontend.
+
+As the name implies: A responsibility of TSFE
+is page rendering. It also handles reading from and writing to the page cache.
+For more details it is best to look into the source code.
+
+There are several contexts in which the term TSFE is used:
+
+*   PHP: It was and is available as global array :php:`$GLOBALS['TSFE']` in PHP.
+*   TypoScript: TypoScript function :ref:`TSFE <t3tsref:data-type-gettext-tsfe>`
+    which can be used to access public properties in TSFE.
+*   (deprecated since v9.5 and :doc:`removed in 10.0
+    <ext_core:Changelog/10.0/Breaking-88564-PageTSconfigSettingTSFEconstantsRemoved>`)
+    Page TSconfig: :typoscript:`TSFE.constants`.
+
+The TypoScript part is covered in the
+:ref:`TypoScript Reference: TSFE <t3tsref:data-type-gettext-tsfe>`.
+In this section we focus on the PHP part and give an overview, in which way the
+TSFE class can be used.
+
+Accessing TSFE
+===============
+
+.. attention::
+
+    Some of the former public properties and methods have been changed to
+    protected or marked as internal. Often, accessing TSFE is no longer
+    necessary, and there are better alternatives.
+
+    Access :php:`$GLOBALS['TSFE']` directly only as a last resort,
+    usage is strongly discouraged, if not absolutely necessary.
+
+From the source:
+
+    When calling a frontend page, an instance of this object is available
+    as :php:`$GLOBALS['TSFE']`, even though the Core development strives to get
+    rid of this in the future.
+
+TSFE is not available in all contexts. In particular, it is
+only available in frontend contexts, not in the backend or CLI.
+
+Initializing :php:`$GLOBALS['TSFE']` in the backend is sometimes done in code
+examples found online. This is not recommended. TSFE is not initialized in the
+backend context by the Core (and there is usually no need to do this).
+
+From the PHP documentation:
```

Comment:
I assume, this can be removed, as TYPO3 v10 is only PHP until 7.4.

Reply:
done